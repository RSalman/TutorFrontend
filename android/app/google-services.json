{
  "project_info": {
    "project_number": "779865862095",
    "firebase_url": "https://prospr-8a50d.firebaseio.com",
    "project_id": "prospr-8a50d",
    "storage_bucket": "prospr-8a50d.appspot.com"
  },
  "client": [
    {
      "client_info": {
        "mobilesdk_app_id": "1:779865862095:android:1b83d4b60becf7dd",
        "android_client_info": {
          "package_name": "com.tutorfrontend"
        }
      },
      "oauth_client": [
        {
          "client_id": "779865862095-4tk4rhhj654t9tepp8c6i5elqlaj6sgg.apps.googleusercontent.com",
          "client_type": 3
        }
      ],
      "api_key": [
        {
          "current_key": "AIzaSyBDjtDNEYE5WbvPy3VumSoq1dEDhKFcOH8"
        }
      ],
      "services": {
        "analytics_service": {
          "status": 1
        },
        "appinvite_service": {
          "status": 1,
          "other_platform_oauth_client": []
        },
        "ads_service": {
          "status": 2
        }
      }
    }
  ],
  "configuration_version": "1"
}