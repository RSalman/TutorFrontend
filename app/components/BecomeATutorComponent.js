import React, { Component } from 'react';
import { StyleSheet, View, ActivityIndicator } from 'react-native';
import { Button } from 'react-native-elements';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { Actions } from 'react-native-router-flux';
import StyledText from './StyledText';
import { submitForm, setProgressBar } from '../actions/signup';
import ErrorView from './ErrorView';

class BecomeATutorComponent extends Component {
  constructor(props) {
    super(props);
  }

  componentWillMount() {
    this.props.setProgressBar(0.75);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.successfulSubmission)
      Actions.tutors({ type: 'reset' });
  }

  navigateToTutorForm() {
    Actions.tutorForm();
  }

  renderErrorView() {
    if (this.props.error) {
      return (
        <ErrorView error={this.props.error} color={brightRed} style={errorStyle} />
      );
    }
  }

  renderSpinner() {
    if (this.props.isLoading) {
      return (
        <ActivityIndicator
          animating
          style={styles.spinner}
          size="large"
          color={buttonColor}
        />
      );
    }
  }

  render() {
    return (
      <View style={styles.container}>
        <StyledText style={styles.headerText}>Would you like to become a tutor?</StyledText>
        <View style={styles.buttonContainer}>
          <Button raised title="Yes" onPress={() => this.navigateToTutorForm()} backgroundColor={buttonColor} />
          <Button raised title="No" onPress={() => this.props.submitForm(this.props.signup_data)} backgroundColor={grey} />
        </View>
        { this.renderErrorView() }
        <View style={styles.spinnerContainer}>
          { this.renderSpinner() }
        </View>
      </View>
    );
  }
}

/* Define colours */
const grey = '#bdc6cf';
const brightRed = '#ff7575';
const white = '#FFF';
const buttonColor = '#FF3366';

const errorStyle = { paddingTop: 80 };
const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignSelf: 'center',
    marginTop: 60,
    padding: 10,
    marginLeft: 15,
    marginRight: 15
  },
  headerText: {
    fontSize: 20,
    textAlign: 'center',
    color: white
  },
  buttonContainer: {
    alignSelf: 'center',
    marginTop: 10,
    flexDirection: 'row'
  },
  spinnerContainer: { paddingTop: 80 },
  spinner: { height: 18 }
});

const mapStateToProps = (state) => {
  return {
    successfulSubmission: state.signup.successfulSubmission,
    signup_data: state.signup.signup_data,
    error: state.signup.error,
    isLoading: state.signup.isLoading
  };
};

const mapDispatchToProps = (dispatch) => bindActionCreators({ submitForm, setProgressBar }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(BecomeATutorComponent);
