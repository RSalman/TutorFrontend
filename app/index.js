import React, { Component } from 'react';
import { StyleSheet,TouchableHighlight, Text, View, Navigator } from 'react-native';

import HomeView from './views/HomeView';
import TutorsComponent from './components/TutorsComponent';
import LeilaFeatureView from './views/LeilaFeatureView';
import SalmanFeatureView from './views/SalmanFeatureView';
import MuraadFeatureView from './views/MuraadFeatureView';

export default class TutoringApp extends Component {
  render() {
    return (
  		<Navigator
        	initialRoute={{screen: 'HomeView'}}
        	renderScene={(route, nav) => {return this.renderScene(route, nav)}}  />
    );
  }

 renderScene(route,nav) {
    switch (route.screen) {
      	case "HomeView":
        	return <HomeView navigator={nav} />
      	case "TutorsComponent":
        	return <TutorsComponent navigator={nav} />
      	case "SalmanFeature":
        	return <SalmanFeatureView navigator={nav} /> 
        case "LeilaFeature":
        	return <LeilaFeatureView navigator={nav} /> 
        case "MuraadFeature":
        	return <MuraadFeatureView navigator={nav} /> 
      }
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  }
});
