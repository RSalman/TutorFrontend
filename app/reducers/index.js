import { combineReducers } from 'redux';
import tutors from './tutors';
import profile from './profile';

const rootReducer = combineReducers({
  tutors: tutors,
  profile: profile
});

export default rootReducer;
