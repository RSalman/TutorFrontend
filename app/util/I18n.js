import I18n from 'react-native-i18n';

import en from './locales/en';
import fr from './locales/fr';

I18n.fallbacks = true;

I18n.translations = {
  en,
  fr
};

global.I18n = I18n;
