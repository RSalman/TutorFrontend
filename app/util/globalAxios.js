import axios from 'axios';
import { Alert, NetInfo } from 'react-native';

axios.defaults.baseURL = 'http://cpserver.eastus.cloudapp.azure.com:2000/api/v1';
axios.defaults.headers.post['Content-Type'] = 'application/json';
axios.defaults.headers.get['Content-Type'] = 'application/json';
axios.defaults.headers.post.Accept = 'application/json';
axios.defaults.headers.get.Accept = 'application/json';

var connection = true;
function connectivityChange(isConnected) {
  connection = isConnected;
}

function showNoInternetDialog(config) {
  return new Promise(function(resolve, reject) {
    Alert.alert(
      I18n.t('globalWifiDialog.noInternetConnection'),
      I18n.t('globalWifiDialog.fixInternet'),
      [
        { text: I18n.t('retry'), onPress: () => resolve(axios(config)) },
      ],
      { cancelable: false }
    );
  });
}

NetInfo.isConnected.addEventListener('change', connectivityChange);

axios.interceptors.response.use(undefined, function(error) {
  return connection ? Promise.reject(error) : showNoInternetDialog(error.config);
});

global.axios = axios;
