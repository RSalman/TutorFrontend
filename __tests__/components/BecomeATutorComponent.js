import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import BecomeATutorComponent from '../../app/components/BecomeATutorComponent';
import '../../app/index';
import '../../app/util/globalAxios';
import '../../app/util/I18n';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

function setup(state) {
  const store = mockStore(state);
  const props = {
    setProgressBar: jest.fn(),
    store
  };
  return props;
}

describe('<BecomeATutorComponent />', () => {
  it('renders correctly', () => {
    const props = setup({
      signup: {
       phoneVerified: false,
        phoneVerifying: false,
        signup_data: {
          first_name: 'John',
          last_name: 'Doe',
          email: 'jdoe@uottawa.ca',
          password: '',
          phone_number: ''
        }
      }
    });
    const tree = renderer.create(
      <BecomeATutorComponent {...props} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
