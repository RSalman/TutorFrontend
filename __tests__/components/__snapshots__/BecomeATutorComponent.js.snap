exports[`<BecomeATutorComponent /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "justifyContent": "center",
      "marginLeft": 15,
      "marginRight": 15,
      "marginTop": 60,
      "padding": 10,
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "fontFamily": "HelveticaNeue-Thin",
        },
        Object {
          "color": "#FFF",
          "fontSize": 20,
          "textAlign": "center",
        },
      ]
    }>
    Would you like to become a tutor?
  </Text>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flexDirection": "row",
        "marginTop": 10,
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "marginLeft": 15,
              "marginRight": 15,
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "#FF3366",
            },
            undefined,
            Object {
              "shadowColor": "rgba(0,0,0, .4)",
              "shadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "shadowOpacity": 1,
              "shadowRadius": 1,
            },
            Object {
              "padding": 12,
            },
            undefined,
            undefined,
            undefined,
          ]
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": undefined,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }>
          Yes
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "marginLeft": 15,
              "marginRight": 15,
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "#bdc6cf",
            },
            undefined,
            Object {
              "shadowColor": "rgba(0,0,0, .4)",
              "shadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "shadowOpacity": 1,
              "shadowRadius": 1,
            },
            Object {
              "padding": 12,
            },
            undefined,
            undefined,
            undefined,
          ]
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": undefined,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }>
          No
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingTop": 80,
      }
    } />
</View>
`;
